indico/modules/rb_new/client/js/components/TimeRangePicker.jsx

```diff
@@ -0,0 +1,203 @@
+        while (next < end) {
+            duration = _humanizeDuration(moment.duration(next.diff(start)));
+            serializedNext = _serializeTime(moment(next));
+            text = <span>{serializedNext} <span styleName="duration">({duration })</span></span>;
+            options.push({key: serializedNext, value: serializedNext, text});
+            next.add(30, 'm');
+        }
+        return options;
+    };
+
+    addOptionIfNotExist = (options, value) => {
+        const found = options.some((el) => {
+            return el.value === value;
+        });
+        if (!found) {
+            options.unshift({key: value, value, text: value});
+            return true;
+        }
+        return false;
+    };
+
+    updateStartTime = (startTime, endTime, duration) => {
+        const start = moment(startTime, ['HH:mm', 'HHmm', 'Hmm']);
+        let end = toMoment(endTime, 'HH:mm');
+        if (start.isValid()) {
+            if (end <= start) {
+                end = moment(start).add(duration);
+                if (end > moment().endOf('day')) {
+                    end = moment().endOf('day');
+                }
+            } else {
+                duration = moment.duration(end.diff(start));
+            }
+            const startOptions = this.generateStartTimeOptions();
+            const endOptions = this.generateEndTimeOptions(start);
+            this.addOptionIfNotExist(startOptions, _serializeTime(moment(start)));
+            this.addOptionIfNotExist(endOptions, _serializeTime(moment(end)));
+            this.setState({
+                startTime: start,
+                endTime: end,
+                duration,
+                startOptions,
+                endOptions
+            });
+            const {onChange} = this.props;
+            onChange(start, end);
+        }
+    };
+
+    updateEndTime = (startTime, endTime, duration) => {
+        let start = toMoment(startTime, 'HH:mm');
+        const end = moment(endTime, ['HH:mm', 'HHmm', 'Hmm']);
+        if (end.isValid()) {
+            if (end <= start) {
+                start = moment(end).subtract(duration);
+                if (start < moment().startOf('day')) {
+                    start = moment().startOf('day');
+                }
+            } else {
+                duration = moment.duration(end.diff(start));
+            }
+            const startOptions = this.generateStartTimeOptions();
+            const endOptions = this.generateEndTimeOptions(start);
+            this.addOptionIfNotExist(startOptions, _serializeTime(moment(start)));
+            this.addOptionIfNotExist(endOptions, _serializeTime(moment(end)));
+            this.setState({
+                startTime: start,
+                endTime: end,
+                duration,
+                startOptions,
+                endOptions,
+            });
+            const {onChange} = this.props;
+            onChange(start, end);
+        }
+    };
+
+    searchTime = () => {
```

Comment:
what is the point of this function when passed to `Dropdown`?

Reply:
It overrides the default behaviour of search, so when you start typing you get no options

Reply:
yeah, yeah, but isn't it enough to not include the `search` attribute in the first place? or at least do it as `search={true}`. I don't understand why you pass the `search` prop even if you don't need the searching functionality

Reply:
So basically including search property let's you type in the field and it's required by addition properties. Keeping the default behavior of search would try to match current options with what you type, so if you started typing e.g. '1' you'd get all the time options that include 1 which doesn't make much sense in a time picker. That's why I need this method to make sure that whatever you type, you get no options as a result of search.

Reply:
Ahh, that is what I didn;t understand - `search` being required by some other properties. However you can make it an inline arrow function `search={() => []}`

Reply:
that's a good idea :)